packages/abstract-signer/src.ts/index.ts

```diff
@@ -229,7 +229,7 @@ export abstract class Signer {
 
         try{
             const response = await hederaTx.execute(this.provider.getHederaClient());
-            return hexStripZeros(response.bytes);
+            return hexlify(response.bytes);
```

Comment:
I am curious to know why we would not strip zeros (cc @joan41868 ) and the reason behind this change

Reply:
`buildCall` is exporting the tx result via `contract.interface.decodeFunctionResult`, under the hood `decodeFunctionResult` has that piece of code:
```
switch (bytes.length % this._abiCoder._getWordSize()) {
    case 0:
        try {
            return this._abiCoder.decode(functionFragment.outputs, bytes);
        } catch (error) { }
        break;

    case 4: {
        const selector = hexlify(bytes.slice(0, 4));
        const builtin = BuiltinErrors[selector];
        if (builtin) {
            errorArgs = this._abiCoder.decode(builtin.inputs, bytes.slice(4));
            errorName = builtin.name;
            errorSignature = builtin.signature;
            if (builtin.reason) { reason = errorArgs[0]; }
        } else {
            try {
                const error = this.getError(selector);
                errorArgs = this._abiCoder.decode(error.inputs, bytes.slice(4));
                errorName = error.name;
                errorSignature = error.format();
            } catch (error) {
                console.log(error);
            }
        }
        break;
    }
}

return logger.throwError("call revert exception", Logger.errors.CALL_EXCEPTION, {
    method: functionFragment.format(),
    errorArgs, errorName, errorSignature, reason
});
```

If we didn't make a `hexlify`, the `bytes.length` value is dynamic and the exception has been thrown, with `hexlify` we always receive a 32 as `bytes.length` that matches the `_getWordSize()`, and handled by `case 0`.

Reply:
Okay, got it!